### PR TITLE
Add credentials token for browser IDE

### PR DESCRIPTION
**Pull request by mustard-mh**
## Description

Part 1/2

## Related Issue(s)

Fixes #
Related https://github.com/gitpod-io/security/issues/121

## How to test

Should test in https://github.com/gitpod-io/gitpod/pull/16404

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-slow-database
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Review by gtsiolis — APPROVED** [review]
Approving to unblock merging, but holding for `@gitpod-io/engineering-webapp`.

/hold

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (mustard-mh) [inline components/server/src/workspace/gitpod-server-impl.ts:695]: > we already do the same for pinning for instance: https://github.com/gitpod-io/gitpod/blob/55f5845dee239d357d96473decb7105e4166bcf7/components/server/src/workspace/gitpod-server-impl.ts#L854-L882

source https://github.com/gitpod-io/gitpod/commit/f099a51c382a36e9b7fbb57f730e8cc58ba88377#r100569735

File: components/gitpod-protocol/src/frontend-dashboard-service.ts
Comment (geropl) [inline components/gitpod-protocol/src/frontend-dashboard-service.ts:49]: @mustard-mh I don't have a good overview over the structure of the new IDE service. But from the past I can tell that mixing "Status" and "token" always ends up problematic. What about introducing a new method "getCredentialsToken" (or similar) on the interface (I cannot tell which is which just by the names "Server" and "Client")?

File: components/gitpod-protocol/src/gitpod-service.ts
Comment (geropl) [inline components/gitpod-protocol/src/gitpod-service.ts:120]: :+1:

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (geropl) [reply components/server/src/workspace/gitpod-server-impl.ts:695]: :+1:

**Review by geropl — APPROVED** [review]
WebApp parts LGTM.

Raising [this concern](https://github.com/gitpod-io/gitpod/pull/16382#discussion_r1106702633), but as this is IDE code I leave it up to you how to go about it. :balance_scale:

File: components/gitpod-protocol/src/frontend-dashboard-service.ts
Comment (mustard-mh) [reply components/gitpod-protocol/src/frontend-dashboard-service.ts:49]: Maybe it's a naming problem as `Status` can be `Info` or anything else, it's a define of interface that `supervisor/frontend` is needed. 

🙏 I will update PR to change the name with compatibility (first we will keep them (old and new name) both, but will remove it later)

Comment (mustard-mh) [reply components/gitpod-protocol/src/frontend-dashboard-service.ts:55]: [Addressed](https://github.com/gitpod-io/gitpod/compare/fa01d4504e464b05a8a3df5e55ac7f62186bfff9..913fa7145fa1f591022c3ec6d9b5d8d92782df00), going to do compatibility test in another PR, will post results here and unhold this PR if it works

**Comment by mustard-mh** [comment]
image

/unhold

**Pull request merged**
